src/sbomber.py

```
@@ -111,8 +113,11 @@ def _download_charm(artifact: Artifact) -> str:
 
 def _download_snap(artifact: Artifact) -> str:
     """Download a snap from the snap store."""
-    cmd = _download_cmd("snap", artifact)
-    proc = subprocess.run(cmd, capture_output=True, text=True)
+    cmd = _download_cmd("snap", artifact, artifact.snap)
+    env = {}
+    if artifact.arch:
+        env["UBUNTU_STORE_ARCH"] = artifact.arch
```

let's raise an error if this is set but the artifact isn't a deb.